torch/csrc/jit/passes/quantization.cpp

```diff
@@ -580,8 +629,20 @@ TORCH_API void InsertObservers(
     script::Module& module,
     const std::string& method_name,
     const QConfigDict& qconfig_dict) {
-  auto module_qconfig_map = getQConfigMap(module, qconfig_dict);
-  InsertObserversImpl(module, method_name, module_qconfig_map);
+  ModuleQConfigMap module_qconfig_map;
+  // Set of child modules that has the same qconfig
+  // as parent module, used for avoiding inserting
```

Comment:
Why can't we just add these children to the map we're generating (so that `qmap.at(child)` == `qmap.at(parent)`)? Having an extra set seems unnecessary and also confusing, I had to reread the code and the comment several times and I'm still not sure I understood it correctly.

Reply:
the reason is we can't compare QConfig, it's a tuple of two script::Module and we don't have a way to compare script::Modules I think. That's why I record all the child modules that inherit the qconfig of parent in the place when we are filling the QConfigMap.